sound/soc/sof/pcm.c

```diff
@@ -299,22 +299,26 @@ static int sof_pcm_hw_free(struct snd_soc_component *component,
 	dev_dbg(component->dev, "pcm: free stream %d dir %d\n",
 		spcm->pcm.pcm_id, substream->stream);
 
+	cancel_work_sync(&spcm->stream[substream->stream].period_elapsed_work);
```

Comment:
now wondering if this should be move to after the 'stop DMA'?

Isn't there a chance that we get a stream interrupt after this cancel_work_sync() and before stopping the DMA?

Reply:
my earlier comment is still valid, I am not longer sure where we should stop this workqueue

Reply:
@plbossart I'm not sure how much of a difference cancelling the workqueue before or after stopping DMA makes. In either case, the pipeline has already been stopped right?

Reply:
@plbossart I've now moved the cancel to the end. I'm still not sure if it is correct but at least it aligns cases where a hw_free is called after a stop or without a stop.
In case of a normal stop, pcm_free, dma_stop and widget_list_free would already have been done before we get to hw_free() which really only needs to cancel the workqueue.